include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -214,14 +220,7 @@ destroy(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __
 
     if constexpr (!::std::is_trivially_destructible_v<_ValueType>)
     {
-        using _ExecutionPolicyDest =
-#if (_PSTL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN || _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN)
-            typename EvaDestroylWorkaroundPolicy<::std::decay_t<_ExecutionPolicy>>::ExecutionPolicy;
-#else
-            _ExecutionPolicy;
-#endif // _PSTL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN || _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN
-
-        const auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(__exec, __first);
+        const auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(get_destroy_dest_policy(__exec), __first);
```

Comment:
1. `get_destroy_dest_policy` is not proper name for such function, IMHO.
2. In case we would not have a broken configuration we will not have `get_destroy_dest_policy` at all, right? But the proposed change introduces such a "policy type converter" always. I think the in-placed code differentiation via preprocessor macro is normal approach and we dont need introduce any new helper stuff like "policy type converter."

Reply:
Which function name would you like to propose?